search/includes/classes/class-search.php

```diff
@@ -339,6 +339,15 @@ public function filter__ep_feature_active( $active, $feature_settings, $feature
 	 * constant should be set to `true`, which will enable query integration for all requests
 	 */
 	public static function ep_skip_query_integration( $skip, $query = null ) {
+		// If the query count has exceeded the maximum
+		//     Only allow half of the queries to use VIP Search
+		if ( self::query_count_incr() > self::$max_query_count ) {
```

Comment:
One thought I had: should we only trigger this when we determine `$skip === false`? In contexts where we return `true`, and ES doesn't kick in, we're doing unnecessary memcache operations and incorrectly incrementing rate limit counts.

Reply:
True, the skip check should be at the top 👍

Reply:
I think what Mo was getting at is it would be better to run all the logic in this function first (remove the early return on the `$_GET['es']` check), and _then_ only rate limit if `$skip === false`.

Currently, we check the counter even if the rest of the filter ultimately returns `true` and skips the integration (if the option or constant is not set).

Reply:
We could add the rate limit as a separate late running (PHP_INT_MAX) filter on `es_skip_query_integration` to solve those concerns - let this filter callback run normally, then our rate limit filter callback runs at the very end, disabling the query integration if we're rate limited (and query integration had been enabled via other filters).

Reply:
> We could add the rate limit as a separate late running (PHP_INT_MAX) filter on es_skip_query_integration to solve those concerns

Ah, that a great idea. Much cleaner and easier to disable / detangle if we ever need to.

Reply:
Sounds good to me :+1: